### ISSUE TITLE
GSEA function (clusterProfiler R package)

### ISSUE THREAD

**Issue by msuffiot** [GuangchuangYu/clusterProfiler#55]
Hi Guangchuang Yu,

**I would like to perform enrichment analysis on single cell data using GSEA function with 4 classes.**
gsea.out = GSEA(geneList = data[,1],TERM2GENE=term2gene,pvalueCutoff =1,minGSSize=5,exponent= 1)

where:
data[,1], is ranked according to the expression values
CD3D     CD3G     CD27     SIT1    CEP68 CD27-AS1 
34.69430 32.17648 31.58424 26.38844 17.55915 17.53385 

and term2gene, look as follow
  term   name
1 ILC1   CCL5
2 ILC1   CD28
3 ILC1   CD3E
4 ILC1 GRAMD3
5 ILC1    CD7
6 ILC1  SIDT1
...

**The problem is that I have that error and I don't understand why. I also tried to find the whole code for GSEA to get more information about that error but I can not find that, could you help me?**

preparing geneSet collections...
[1] "calculating observed enrichment scores..."
[1] "calculating permutation scores..."
  |                                                                                                           |   0%
[1] "calculating p values..."
Error in pi0est(p, ...) : 
  ERROR: The estimated pi0 <= 0. Check that you have valid p-values or use a different range of lambda.

Thank you very much in advance
Madeleine

**Comment by GuangchuangYu** [comment]
make sure you are using the latest release version of clusterProfiler by using:
```{r}
# install.packages('rvcheck')
require(rvcheck)
check_bioc("clusterProfiler")
```

Then follow the [guideline](https://github.com/GuangchuangYu/clusterProfiler/blob/master/CONTRIBUTING.md) to make a reproducible example.

**Comment by msuffiot** [comment]
Is it possible that the score distribution has an influence? When I used as scores the logFC, which are normally distributed the function works, but when I use the expression values (of single cells) that are not normally distributed (skewed on the left, with high number of negative values) I have that error. So I just added +3 in order to remove negative values and the function worked.

Thank you very very much for you answer
Madeleine
